src/jit/flowgraph.cpp

```diff
@@ -22995,6 +22995,7 @@ void Compiler::fgRemoveEmptyTry()
                     fgRemoveStmt(block, finallyRet);
                     block->bbJumpKind = BBJ_ALWAYS;
                     block->bbJumpDest = continuation;
+                    fgAddRefPred(continuation, block);
                 }
```

Comment:
Is this related to this change?

It looks like preds haven't been computed by the time your try/finally optimization phases are run, so perhaps you should remove all this pred update stuff, and add (for clarity) the following at the top of each phase:
```
noway_assert(!fgComputePredsDone); 
```